Automattic/woocommerce-payments
# HACK week: Display server error messages on Settings save

Fixes #7090

#### Changes proposed in this Pull Request
This PR improves the handling and displays the server errors on the WooPayments settings page. It acts like a second layer of error handling for account update errors that are not covered by front-end or back-end validation. The server errors are rarely a case, but when they happen, the save settings action is displayed as successful, however, the settings are not preserved.

<img width="535" alt="Screenshot 2023-09-07 at 11 29 11" src="https://github.com/Automattic/woocommerce-payments/assets/8667118/47a08774-7d8d-4a06-a3f6-efa24635f3a5">

#### Testing instructions
Testing the server errors is tricky as we validate the account inputs in both the front-end and back-end. However, with minor code modifications, we could bypass the validations.

*Case 1*
- Open the [following file](https://github.com/Automattic/woocommerce-payments/blob/12ac12d4cfacabc8edc9a45ba362c670fa3a9397/includes/class-wc-payment-gateway-wcpay.php#L2600) and on line 2600 add the following code `return $value;`
- Navigate to `Payments` -> `Settings`.
- Enter `Test` in the `Customer bank statement`.
- Save settings.
- Verify the `Error: The statement descriptor matches a common term or website URL, and can't be used.` error is displayed.

*Case 2*
- Open the [following file](https://github.com/Automattic/woocommerce-payments/blob/12ac12d4cfacabc8edc9a45ba362c670fa3a9397/includes/class-wc-payment-gateway-wcpay.php#L2600) and on line 2600 add the following code `return $value;`
- Navigate to `Payments` -> `Settings`.
- Enter `Проба` in the `Customer bank statement`.
- Save settings.
- Verify the `Error: The statement descriptor must contain at least one Latin character.` error is displayed.

*Case 3*
- Remove all modifications on the [file](https://github.com/Automattic/woocommerce-payments/blob/12ac12d4cfacabc8edc9a45ba362c670fa3a9397/includes/class-wc-payment-gateway-wcpay.php#L2600) 
- Navigate to `Payments` -> `Settings`.
- Enter `Проба` in the `Customer bank statement`.
- Save settings.
- Verify no server error is displayed and inline error is displayed.

<img width="507" alt="Screenshot 2023-09-07 at 11 37 16" src="https://github.com/Automattic/woocommerce-payments/assets/8667118/a80e801a-2e94-4f57-91c7-c8f34e30015f">

-------------------


**Post merge**
